client/tests/simulator_tests.rs

```diff
@@ -83,6 +83,63 @@ fn test_get_capability_manufacturer_id() {
     assert_eq!(received, expected);
 }
 
+#[test]
+fn test_get_random() {
+    const REQUEST_SIZE: u16 = 32;
+    const MIN_RECEIVED: u16 = 32;
+
+    let (_sim_lifeline, mut tpm) = get_started_tpm();
+
+    let command = GetRandomCmd {
+        bytes_requested: REQUEST_SIZE,
+    };
+
+    // We allow panic in test cases.
+    let resp = run_command(&command, &mut tpm).expect("Failed running command.");
+
+    // Validate that retrieved data contains at least 32 values (SHA256 digest),
+    // and is less that requested amount.
+    assert!(resp.random_bytes.size >= MIN_RECEIVED);
+    assert!(resp.random_bytes.size <= REQUEST_SIZE);
+
+    // Lets pull out the actual data as a slice for convenience
+    let random_slice = &resp.random_bytes.buffer[0..resp.random_bytes.size as usize];
+
+    // Print for test inspection.
+    println!("Got random sequence: {:?}", random_slice);
+
+    // Crude randomness test. Confirm we did not get identical data sequences.
+    {
+        let mut zeros: u16 = 0;
+        let mut count: [u16; 16] = [0; 16]; // 16 slots.
+        let mut last_value = random_slice[random_slice.len() - 1];
+        let mut i = 0;
+        for i in 0..random_slice.len() as usize {
```

Comment:
It might be slightly nicer to use iterators to get your `prev` and `next` instead of relying on indexing the slice:

```rs
for (prev, curr) in random_slice.iter().zip(random_slice.iter().cycle().skip(1)) {
    // Your calculation.
}
```

Though I'm not sure if my brain is just broken from doing too much work with iterators. So take this as an optional. Note that it doesn't technically iterate in the exact same order, but I think it accomplishes the same thing, as it iterates over all pairs of `(prev, curr)` (the difference is the pair `([len()-1],[0])` happens at the end in the iterator construction above).

Reply:
I love this Rust construct 👍 I played a little around with it 🙂 The zip will only provide a value pair iterator corresponding to the shorter of the two iterators. Cycle() loops the one iterator, so it infinitely wraps around. And skip offsets the iterator 1 element in this case. In the simplified crude test, I just need adjacent value pairs, so I adopted this one and skipped the cycle 👍